api/src/main/java/com/cloud/network/element/NetworkElement.java

```diff
@@ -87,6 +88,12 @@ boolean prepare(Network network, NicProfile nic, VirtualMachineProfile vm, Deplo
     boolean release(Network network, NicProfile nic, VirtualMachineProfile vm, ReservationContext context) throws ConcurrentOperationException,
         ResourceUnavailableException;
 
+    /**
+     * Release IP from the network provider if reserved
+     * @param ipAddress
+     */
+    boolean releaseIp(IpAddress ipAddress);
```

Comment:
```suggestion
    default boolean releaseIp(IpAddress ipAddress) {
        return true;    
    }
```

have this default implementation (don't override wherever not required)